unittests/platformagnostictests.py

```diff
@@ -336,7 +336,41 @@ def test_editorconfig_match_path(self):
 
         for filename, pattern, expected in cases:
             self.assertTrue(match_path(filename, pattern) is expected, f'{filename} -> {pattern}')
-    
+
+    def test_format_invalid_config_key(self) -> None:
+        fd, fname = tempfile.mkstemp(suffix='.ini', text=True)
+        self.addCleanup(os.unlink, fname)
+
+        with os.fdopen(fd, 'w', encoding='utf-8') as handle:
+            print('not_an_option = 42', file=handle)
```

Comment:
why use `os.fdopen` and not just `open` on the `fname`?

Reply:
Maybe I need to think about this more, I thought `os.fdopen` simply wraps the underlying `fd` in a `file` object, but that it doesn't close it, simply invalidates the `file` object. I'll have to poke at it and see if that understanding is incorrect.

Reply:
according to the documentation (https://docs.python.org/3/library/os.html#os.fdopen):

> Return an open file object connected to the file descriptor fd. This is an alias of the [open()](https://docs.python.org/3/library/functions.html#open) built-in function and accepts the same arguments. The only difference is that the first argument of [fdopen()](https://docs.python.org/3/library/os.html#os.fdopen) must always be an integer.

Therefore, I assume that the context manager effectively closes the file.